pkg/rsyslog/rsyslog.conf

```diff
@@ -59,14 +59,16 @@ if $parsesuccess == "FAIL" then {
                Target="127.0.0.1"
                Port="5140"
                Protocol="tcp"
-               template="nonJsonOutput")
+               template="nonJsonOutput"
+               queue.size="1000" queue.type="LinkedList")
```

Comment:
Does linkedlist reduce the risk of corruption?

Reply:
It does somewhat (provided that messages are send to the cloud quickly enough). But I implemented it here mostly to guarantee that messages always end up in syslog.txt even if the logmanager is down and the action blocks

Reply:
Rsyslogd starts before logmanager. There is a windows when rsyslogd tries but fails making TCP connection to logmanager. This change makes the initial set of logs (onboot logs) be stored in memory (moved from disk to memory). We need to see if these messages are retried for the logmanager branch of actions. File write action would go through anyway.

Reply:
This was done to unblock creation of syslog.txt (if the logmanager action blocks -- the next action doesn't seem to fire). You're right in a sense that we now have a window of messages that can die together with rsyslog if it dies and never reach the cloud. They are, however, guaranteed to be in syslog.txt -- so at least that is good.

That said, @gkodali-zededa -- as part of the immediate improvements in rsyslog reliability I think you will have to look into whether this two needs to be made into an on-disk queue or not.

Definitely more research about how to use rsyslog effectively is very much needed and I hope you can do that.